contracts/strategies/QVImpactStream.sol

```diff
@@ -0,0 +1,205 @@
+                revert RECIPIENT_ERROR(payout.recipientId);
+            }
+
+            payouts[recipientId] = amount;
+            totalAmount += amount;
+            unchecked {
+                ++i;
+            }
+        }
+
+        if (totalAmount > poolAmount) revert PAYOUT_MORE_THAN_POOL_BALANCE();
+
+        emit PayoutSet(_payouts, msg.sender);
+    }
+
+    /// =============================
+    /// ==== Internal Functions =====
+    /// =============================
+
+    function _distribute(address[] memory _recipientIds, bytes memory, address _sender)
+        internal
+        virtual
+        override
+        onlyAfterAllocation
+    {
+        IAllo.Pool memory pool = allo.getPool(poolId);
+        address poolToken = pool.token;
+
+        uint256 length = _recipientIds.length;
+        for (uint256 i = 0; i < length;) {
+            address recipientId = _recipientIds[i];
+            Recipient storage recipient = _recipients[recipientId];
+
+            address recipientAddress = recipient.recipientAddress;
+            uint256 amount = payouts[recipientId];
+
+            if (amount == 0) revert RECIPIENT_ERROR(recipientId);
+
+            delete payouts[recipientId];
+
+            _transferAmount(poolToken, recipientAddress, amount);
+
+            bytes memory data = abi.encode(recipientAddress, amount, _sender);
+
+            emit Distributed(recipientId, data);
+
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    /// =========================
+    /// ==== View Functions =====
+    /// =========================
+
+    /// @notice Get the total votes received for a recipient
+    /// @param _recipient The address of the recipient
+    /// @return The total votes received by the recipient
+    function getTotalVotesForRecipient(address _recipient) external view returns (uint256) {
+        return _votingState.recipientVotes[_recipient];
+    }
+
+    /// @notice Get the payout for a single recipient
+    /// @param _recipientId The ID of the recipient
+    /// @return The payout as a 'Payout' struct
+    function getPayout(address _recipientId) external view returns (Payout memory) {
+        uint256 amount = payouts[_recipientId];
+        return Payout(_recipientId, amount);
+    }
+
+    /// @notice Transfer the funds recovered  to the recipient
+    /// @dev 'msg.sender' must be pool manager
+    /// @param _token The token to transfer
+    /// @param _recipient The recipient
+    function recoverFunds(address _token, address _recipient) external onlyPoolManager(msg.sender) {
+        // Get the amount of the token to transfer, which is always the entire balance of the contract address
+        uint256 amount = _token == NATIVE ? address(this).balance : IERC20Upgradeable(_token).balanceOf(address(this));
+
+        // Transfer the amount to the recipient (pool owner)
+        _transferAmount(_token, _recipient, amount);
+    }
```

Comment:
I was wondering if the `withdraw` function can achieve the same goal.

Reply:
fair, yes we can use withdraw instead